### ISSUE TITLE
property-naming flags extension properties

### ISSUE THREAD

**Issue by vanniktech** [pinterest/ktlint#2024]
The following code gets flagged with 0.49.1:

```kotlin
val Context.adsDelegateFactory @SuppressLint("WrongConstant") get() =
  applicationContext.getSystemService(SYSTEM_SERVICE_AD_DELEGATE_FACTORY) as AdsDelegateFactory
```

> Property name should use the screaming snake case notation when the value can not be changed (standard:property-naming)

I would not expect this, if you're adding a property via an extension and leveraging `get()` since the value can indeed change inside the implementation.

**Comment by vanniktech** [comment]
So when I remove `@SuppressLint("WrongConstant")` it does not get flagged as expected. So there's something weird going on.

**Comment by paul-dingemans** [comment]
Indeed, the annotation causes the check to fail. Same happens with:
```
val Context.adsDelegateFactory private get() =
    applicationContext.getSystemService(SYSTEM_SERVICE_AD_DELEGATE_FACTORY) as AdsDelegateFactory
```

**Comment by paul-dingemans** [comment]
Problem will either be resolved in `0.49.2` only when more pressing bugs need to be fixed. Otherwise it will be fixed in the next normal release.
